core/src/main/java/hudson/search/Search.java

```diff
@@ -73,7 +76,31 @@ public class Search implements StaplerProxy {
      */
     private static /* nonfinal for Jenkins script console */ int MAX_SEARCH_SIZE = Integer.getInteger(Search.class.getName() + ".MAX_SEARCH_SIZE", 500);
 
-    public void doIndex(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
+    public void doIndex(StaplerRequest2 req, StaplerResponse2 rsp) throws IOException, ServletException {
+        if (Util.isOverridden(Search.class, getClass(), "doIndex", StaplerRequest.class, StaplerResponse.class)) {
+            try {
+                doIndex(StaplerRequest.fromStaplerRequest2(req), StaplerResponse.fromStaplerResponse2(rsp));
+            } catch (javax.servlet.ServletException e) {
+                throw ServletExceptionWrapper.toJakartaServletException(e);
+            }
+        } else {
+            doIndexImpl(req, rsp);
+        }
+    }
+
+    /**
+     * @deprecated use {@link #doIndex(StaplerRequest2, StaplerResponse2)}
+     */
+    @Deprecated
```

Comment:
```suggestion
    @Deprecated
    @StaplerNotDispatchable
```
No need to route it.